crates/data-model/src/tokens.rs

```diff
@@ -294,7 +294,8 @@ impl TokenType {
     pub fn check(token: &str) -> Result<TokenType, TokenFormatError> {
         // these are legacy tokens imported from Synapse
         // we don't do any validation on them and continue as is
-        if token.starts_with("syt_") {
+        // Tokens starting `MDAx` are macaroons from even older versions of Synapse.
```

Comment:
If I'm being pedantic for a second, macaroons are a list of hex length plus (key space value), then HMAC'd and base64'd.

The first K/V is `location`/servername, which means:

 - it will start with `000flocation ` for HS with a single character
 - it be `001Nlocation ` (with N varying) for HS with 2 to 17 characters
 - etc.

So the prefix `MDAx` works because it's the base64 serialisation of `001`. The `MDA` prefix would work for server names up to 145 characters.

If we want to be extra sure, we would:

 - base64url decode the string
 - check on the result that byte 5 to 13 are exactly `6c 6f 63 61 74 69 6f 6e` (`location`)

Reply:
fair for digging in to the format! I wasn't able to find (and still haven't found) the spec for Macaroons.
I probably should have looked at some code at least